[docs] Document known EDOT component matrix issue causing missing `service_destination` metrics in 9.2/9.3 (#14965)
elastic/elastic-agent: docs/release-notes/known-issues.md

For more information, check [Issue #14764](https://github.com/elastic/elastic-agent/issues/14764).
:::

> :::{dropdown} Missing APM `service_destination` metrics and empty service map on APM UI

Reviewer: Unfortunately, the code formatting doesn’t render in the dropdown title, so:

```suggestion
:::{dropdown} Missing APM service_destination metrics and empty service map in the APM UI
```

**Applies to: {{agent}} 9.2.6, 9.2.7, 9.3.1, 9.3.2**

> Aggregated `service_destination` metric used by APM UI may be missing from `metrics/aggregated-otel-metrics` EDOT otel collector pipeline even when trace traffic is present and other telemetry paths appear healthy. This causes APM UI service map to be empty.

Reviewer: ```suggestion
On June 17, 2026, a known issue was discovered where the aggregated `service_destination` metric that the APM UI uses can be missing from the `metrics/aggregated-otel-metrics` EDOT Collector pipeline, even when trace traffic is present and other telemetry paths appear healthy. As a result, the APM service map appears empty.
```

> This issue is caused by the specific EDOT component-version combination of `elasticapmprocessor` + `elasticapmconnector` both on v0.29.0, which is the bundled version in the affected Elastic Agent versions.

Reviewer: We could simplify this a bit?

```suggestion
The issue occurs when two EDOT components, `elasticapmprocessor` and `elasticapmconnector`, are both at v0.29.0, which is the version bundled with the affected {{agent}} releases.
```

Reviewer 2: your edit lgtm. I think having the details of v0.29.0 here is useful because sometimes customers don't use EA directly, but they build a collector from a collection of components.

**Resolution**

> This issue is fixed in {{agent}} 9.2.8 and 9.3.3, where the affected component combinations are no longer present.

Reviewer: ```suggestion
This issue is fixed in {{agent}} 9.2.8 and 9.3.3, which include updated versions of these components.
```

For more information, check [Issue #14764](https://github.com/elastic/elastic-agent/issues/14764).
:::

> :::{dropdown} Missing `service_destination` metrics in specific 9.2/9.3 EDOT component combinations

Reviewer 2: ```suggestion
:::{dropdown} Missing APM `service_destination` metrics and empty service map on APM UI
```

**Applies to: {{agent}} 9.2.6, 9.2.7, 9.3.1, 9.3.2**

> On June 17, 2026, a known issue was discovered where expected aggregated `service_destination` metric series may be missing (or incomplete) from `metrics/aggregated-otel-metrics` even when trace traffic is present and other telemetry paths appear healthy.

Reviewer 2: ```suggestion
Aggregated `service_destination` metric used by APM UI may be missing from `metrics/aggregated-otel-metrics` EDOT otel collector pipeline even when trace traffic is present and other telemetry paths appear healthy. This causes APM UI service map to be empty.
```

This issue affects specific EDOT component-version combinations where traces are ingested and the `elasticapmprocessor` + `elasticapmconnector` flow executes, but destination-derived metric series are not emitted as expected.

The issue was observed with this component signal:

| elasticapmconnector | elasticapmprocessor | elastictraceprocessor | elasticapmintakereceiver | opentelemetry-lib | otelcol |
|---:|---:|---:|---:|---:|---:|

> | v0.29.0 | v0.29.0 | v0.29.0 | v0.29.0 | v0.30.0 | v0.144.0 / v0.145.0 |

Reviewer 2: ```suggestion
This issue is caused by the specific EDOT component-version combination of `elasticapmprocessor` + `elasticapmconnector` both on v0.29.0, which is the bundled version in the affected Elastic Agent versions.
```
